### PR TITLE
Add rocm checkup for check_binary.sh

### PR DESCRIPTION
**Pull request by walterddr**

### CODE REVIEW

**Comment by walterddr** [comment]
@jeffdaily @seemethere

File: check_binary.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ else
   install_root="$(dirname $(which python))/../lib/python${py_dot}/site-packages/torch/"
 fi
 
-if [[ "$DESIRED_CUDA" != 'cpu' ]]; then
+if [[ "$DESIRED_CUDA" != 'cpu' && "$DESIRED_CUDA" != 'rocm3.7' ]]; then
```
Comment (jeffdaily) [inline check_binary.sh:34]:
```suggestion
if [[ "$DESIRED_CUDA" != 'cpu' && "$DESIRED_CUDA" != *"rocm"* ]]; then
```

Comment (jeffdaily) [reply check_binary.sh:34]: For now let's capture all rocm builds, not just 3.7.

**Review by jeffdaily — CHANGES_REQUESTED** [review]

**Comment by jeffdaily** [comment]
The corresponding PR on the pytorch side is https://github.com/pytorch/pytorch/pull/44250.  It is failing to run the `build_libtorch.sh` script because build_amd.py is not run first to hipify the sources.

This log shows a sample failing build:
https://app.circleci.com/pipelines/github/pytorch/pytorch/210639/workflows/72c5caa6-c1dc-4fbc-b092-b95fcac1a3aa/jobs/7276996

Suggested diff for this PR to run the hipify script.

```diff
diff --git a/manywheel/build_libtorch.sh b/manywheel/build_libtorch.sh
index ffd56cf..5b392f2 100644
--- a/manywheel/build_libtorch.sh
+++ b/manywheel/build_libtorch.sh
@@ -110,6 +110,11 @@ else
     export _GLIBCXX_USE_CXX11_ABI=0
 fi

+if [[ "$DESIRED_CUDA" == *"rocm"* ]]; then
+    echo "Calling build_amd.py at $(date)"
+    python tools/amd_build/build_amd.py
+fi
+
 echo "Calling setup.py install at $(date)"

 if [[ $LIBTORCH_VARIANT = *"static"* ]]; then
```

**Review by seemethere — APPROVED** [review]

**Pull request merged**